src/libs/actions/User.ts

```diff
@@ -469,6 +471,56 @@ function triggerNotifications(onyxUpdates: OnyxServerUpdate[]) {
     });
 }
 
+function playSoundForMessageType(pushJSON: OnyxServerUpdate[]) {
+    try {
+        const reportActionsOnly = pushJSON.filter((update) => update.key.includes('reportActions_'));
+        const flatten = flattenDeep(reportActionsOnly.map((update) => Object.keys(update.value).map((key) => update.value[key])));
+        const types = flatten.map((data) => data.originalMessage).filter(Boolean);
+
+        // Someone completes a task
+        if (flatten.find((data) => data.actionName === 'TASKCOMPLETED')) {
+            return playSound('success');
+        }
+
+        // Someone completes a money request
+        if (flatten.find((data) => data.actionName === 'IOU')) {
+            return playSound('success');
+        }
```

Comment:
while this is easy to read its a bit inefficient to eventually loop 8x over the same data set.
If would be simpler if we'd use a for of loop here and in the for loop write a switch or if statement block. This time we'll only iterate over the data once.